src/sklearn_evaluation/plot/classification.py

```diff
@@ -78,23 +79,25 @@ def __init__(self, y_true, y_pred, target_names=None, normalize=False, cm=None):
                 FutureWarning,
                 stacklevel=3,
             )
-
-        self.figure = plt.figure()
-        ax = self.figure.add_subplot()
-
-        if cm is not None and cm is not False:
-            self.cm = cm
-            self.target_names = target_names
-            self.normalize = normalize
-            cmap, ax = _confusion_matrix_init_defaults(cmap=None, ax=ax)
-        else:
-            self.cm = _confusion_matrix(y_true, y_pred, normalize)
-            self.target_names, cmap, ax = _confusion_matrix_validate(
-                y_true, y_pred, target_names, cmap=None, ax=ax
-            )
-            self.normalize = normalize
-
-        _plot_cm(self.cm, cmap, ax, self.target_names, self.normalize)
+        try:
+            self.figure = plt.figure()
+            ax = self.figure.add_subplot()
+
+            if cm is not None and cm is not False:
+                self.cm = cm
+                self.target_names = target_names
+                self.normalize = normalize
+                cmap, ax = _confusion_matrix_init_defaults(cmap=None, ax=ax)
+            else:
+                self.cm = _confusion_matrix(y_true, y_pred, normalize)
+                self.target_names, cmap, ax = _confusion_matrix_validate(
+                    y_true, y_pred, target_names, cmap=None, ax=ax
+                )
+                self.normalize = normalize
+
+            _plot_cm(self.cm, cmap, ax, self.target_names, self.normalize)
+        except ValueError as e:
```

Comment:
Does it ever throws a `ValueError`?
Or why did we wrap it into try/catch?

Reply:
First I changed all the relevant ValueErrors to PloomberValueError.
Then I wrapped code sections that didn't catch these PloomberValueError.

Since it's an experiment, I didn't dive into the error sources (i.e: some of them were raised by numpy or other 3rd party libraries) - I wrapped it with try/catch so no ValueError will be lost.

Reply:
That makes sense. @edublancas thoughts?